### PR TITLE
mpsl: Update dependencies for 54L bsim model

### PR DESCRIPTION
**Pull request by hermabe**
To allow building for the 54L bsim board

### CODE REVIEW

File: mpsl/Kconfig [hunk 1]
```diff
@@ -11,7 +11,7 @@ config MPSL
 	select NRF_802154_MULTIPROTOCOL_SUPPORT if NRF_802154_RADIO_DRIVER
 	select MULTITHREADING_LOCK
 	depends on (SOC_SERIES_BSIM_NRFXX || SOC_SERIES_NRF52X || SOC_NRF5340_CPUNET ||\
-				SOC_NRF54H20_CPURAD || SOC_SERIES_NRF54LX)
+				SOC_NRF54H20_CPURAD || SOC_COMPATIBLE_NRF54LX)
```
Comment (rugeGerritsen) [inline mpsl/Kconfig:14]: Does change requires https://github.com/zephyrproject-rtos/zephyr/commit/fb774fef74a0c3baf80e5435158f11bbfbb0bc3d

Comment (hermabe) [reply mpsl/Kconfig:14]: Yes, didnt realize this wasnt here yet. Will have to fromtree that, given that the upmerge is delayed.

Some more changes are needed in the bsim board to run sdc anyway (Related to `clock`, `psa-rng`, and `bt-hci` dts).

Comment (hermabe) [reply mpsl/Kconfig:14]: Seems all the dts issues have already been fixed upstream

**Comment by hermabe** [comment]
`SOC_SERIES_BSIM_NRFXX `  is already there, no need for this change.

**Pull request closed without merging**